libmattermost.c

```diff
@@ -3659,15 +3664,16 @@ mm_got_history_of_room(MattermostAccount *ma, JsonNode *node, gpointer user_data
 	gint i, len = json_array_get_length(order);
 
 	// do not show updates (such as reactions). only edits, new posts and deletes
-	for (i = len - 1; i >= 0; i--) {
-		const gchar *post_id = json_array_get_string_element(order, i);
-		JsonObject *post = json_object_get_object_member(posts, post_id);
-
-		const gint64 since = mm_get_channel_approximate_view_time(ma, channel);
-		if (json_object_get_int_member(post, "create_at") < since && json_object_get_int_member(post, "edit_at") < since && json_object_get_int_member(post, "delete_at") < since) {
-			json_array_remove_element(order, i);
-		}
-	}
+	// We don't need this now, as in a channel we are fetching last 60 messages where post "create_at" time could be lesser than channel joined time "since"
+	// for (i = len - 1; i >= 0; i--) {
+	// 	const gchar *post_id = json_array_get_string_element(order, i);
+	// 	JsonObject *post = json_object_get_object_member(posts, post_id);
+
+	// 	const gint64 since = mm_get_channel_approximate_view_time(ma, channel->id);
+	// 	if (json_object_get_int_member(post, "create_at") < since && json_object_get_int_member(post, "edit_at") < since && json_object_get_int_member(post, "delete_at") < since) {
+	// 		json_array_remove_element(order, i);
+	// 	}
+	// }

```

Comment:
Can we remove this commented code, if it is not needed anymore?

Reply:
Have kept it intentionally to get it verified in the final PR from the client, because as per our investigation this piece of code is no longer required now, but just to be on the safer side wanted to know if this has any other use as well. If we get no objection in the final PR I'll remove it before merging.